tools/porting/precice-2-to-3

```diff
@@ -0,0 +1,102 @@
+#! /usr/bin/env python
+
+from lxml import etree
+import sys
+import argparse
+
+
+if sys.version_info < (3, 6):
+    raise RuntimeError(
+        "This program requires python 3.6 or later but you attempted to run it"
+        " with python {}.{}".format(
+            sys.version_info.major. sys.version_info.minor
+        ))
+
+
+def isTrue(text):
+    return text.lower() in ["yes", "1", "true", "on"]
+
+
+def quote(text):
+    return f'"{text}"'
+
+
+def parseXML(content):
+    p = etree.XMLParser(recover=True, remove_comments=True)
+    return etree.fromstring(content, p)
+
+
+def parseXMLFile(filename):
+    return parseXML(open(filename, "rb").read())
+
+def portParticipant(p):
+    # use-mesh -> receive-mesh / provide-mesh
+    for um in p.findall("use-mesh"):
+        if isTrue(um.attrib.get("provide", "false")):
+            um.tag = "provide-mesh"
+            um.attrib.pop("provide", None)
+        else:
+            um.tag = "receive-mesh"
+            um.attrib.pop("provide", None)
+
+    # mappings
+    for m in filter(lambda c: c.tag.startswith("mapping"), p.iterchildren()):
+        # scaled-consisitent constraint
+        if m.attrib["constraint"] == "scaled-consisitent":
+            m.attrib["constraint"] = "scaled-consisitent-surface"
+
+        # rbf mappings
+        if m.tag.startswith("mapping:rbf"):
+            rbf = m.tag.removeprefix("mapping:rbf-")
+            if isTrue(attrib.get("use-qr-decomposition", "false")):
+                m.tag = "mapping:rbf-global-iterative"
```

Comment:
I would just go here for `mapping:rbf` (in both branches). That's what we want the users to configure.